Tests/MongoSwiftTests/BSONValueTests.swift

```diff
@@ -27,4 +28,89 @@ final class BSONValueTests: XCTestCase {
         expect(try Binary(data: sixteenBytes, subtype: .uuidDeprecated)).toNot(throwError())
         expect(try Binary(data: sixteenBytes, subtype: .uuid)).toNot(throwError())
     }
+
+    fileprivate func bsonEqual(expectedValue: BSONValue?) -> Predicate<BSONValue> {
```

Comment:
1) AFAIK, no existing predicates require parameter labels. let's follow that style and add `_ `  before `expectedValue`, and just have the syntax be `expect(x).to(bsonEqual(y))`.

2) let's move this into `TestUtils.swift`. It will be useful throughout all of our tests. 

at some point we should update tests to make them less verbose and use this. You'll notice a lot of places where I do something like
```swift
expect(doc["a"] as? Int).to(equal(5))`
```

but now we can do
```swift
expect(doc["a"]).to(bsonEqual(5))
```

Reply:
Great, done in e81efdb.

I'll make a ticket for updating the other parts of the tests using `expect()` and edit this comment to include a link.